### PR TITLE
Add missing relationships from Laravel core and support custom relationships from external packages.

### PR DESCRIPTION
**Pull request by nonetallt**
### Rationale

This pull request addresses the following problems:

1) The current version does not support "through" type relationships. For example, HasManyThrough relation results in a singular relation instead of the correct plural. We should support [all available Laravel relationship methods](https://github.com/laravel/framework/blob/be2ddb5c31b0b9ebc2738d9f37a9d4c960aa3199/src/Illuminate/Database/Console/ShowModelCommand.php#L51).

2) The current version does not support custom relationships from external packages like [BelongsToThrough](https://github.com/staudenmeir/belongs-to-through) or [BelongsToManyThrough](https://gist.github.com/alexweissman/cae6303d2476d18b4a10eeef38919fcf).

### Changes

**Add support for the following core relationship methods**:
- HasManyThrough
- MorphedByMany
- HasOneThrough
- MorphTo

**Add a new internal command** that acts as a wrapper for the Laravel `ShowModelCommand` and enables the customization of the command to print out custom relationships. The name of this command is `model:typer-show`.

**Add configuration for the package**. This allows the user to change the custom relationships via the config. I originally implemented this via a console command option but having to customized complex options like pluralization using a console string would have been confusing and tedious for the user, hence configuration.

### Notes

**IMPORTANT** `Illuminate\Foundation\Console\ShowModelCommand` is moved to `Illuminate\Database\Console\ShowModelCommand` in Laravel 10+. The wrapper command extends the former of the two, since the current master branch of this package uses `"illuminate/database": "^9.45",` as a dependency. If the package version is made laravel 10+ compatible, the parent class namespace should be updated accordingly. 

The current test setup of the package is incomplete so I could only test this manually by requiring my local version via composer. I think there's another pull request already discussing this but the default test script `composer test` does not actually work, even with the dependencies installed.

### CODE REVIEW

**Comment by acidjazz** [comment]
awesome work @nonetallt  - much appreciated

**Comment by tcampbPPU** [comment]
> The current test setup of the package is incomplete so I could only test this manually by requiring my local version via composer. I think there's another pull request already discussing this but the default test script composer test does not actually work, even with the dependencies installed.

If you have any ideas on how to test this as well that would be greatly appreciated! It still a manual process for myself too, and not sure how to exatclty setup testing automation when it depends so much on basically having a standard Laravel app.

File: src/Commands/ShowModelCommand.php [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?php
+
+namespace FumeApp\ModelTyper\Commands;
+
+use Illuminate\Foundation\Console\ShowModelCommand as BaseCommand;
```
Comment (tcampbPPU) [inline src/Commands/ShowModelCommand.php:5]: I am confused, should it be:
```php
use Illuminate\Database\Console\ShowModelCommand as BaseCommand;
```
feel free to update the `lluminate/database` requirement if needed

Comment (nonetallt) [reply src/Commands/ShowModelCommand.php:5]: - For Laravel < 10, the correct command is `Illuminate\Foundation\Console\ShowModelCommand`
- For Laravel 10+, the correct command is `Illuminate\Database\Console\ShowModelCommand`

The current version requirement of `"illuminate/database": "^9.45"` of master is Laravel 9 compatible.

Comment (tcampbPPU) [reply src/Commands/ShowModelCommand.php:5]: any reason not to update:
```
"illuminate/database": "^10.00"
```
does that make it easier?

Comment (tcampbPPU) [reply src/Commands/ShowModelCommand.php:5]: i'd rather just support the latest version and not have to conflict with older (even if older was just recently updated)

Comment (nonetallt) [reply src/Commands/ShowModelCommand.php:5]: > i'd rather just support the latest version and not have to conflict with older (even if older was just recently updated)

In that case we can change the class to `Illuminate\Database\Console\ShowModelCommand` and db version to `"illuminate/database": "^10.00"` as per your suggestion. The database requirement should also be moved from `require-dev` to `require`.

**Comment by acidjazz** [comment]
I'm all for going 10+ moving forward - esp now that this is at a v2.x

File: src/Commands/ShowModelCommand.php (hunk 1 above)
Comment (tcampbPPU) [reply src/Commands/ShowModelCommand.php:5]: ok sounds good, feel free to make that change

Comment (tcampbPPU) [reply src/Commands/ShowModelCommand.php:5]: besides that everything is looking/working great

**Comment by nonetallt** [comment]
The requirements should be fixed now. I upgraded my application to Laravel 10 to make sure there were no conflicts.

**Review by tcampbPPU — APPROVED** [review]

**Pull request merged**